### ISSUE TITLE
Null Reference Exception from popup

### ISSUE THREAD

**Issue by derekantrican** [AvaloniaUI/Avalonia#9089]
A few users encountered a crash of my application when a popup is shown. Here are the details:

```
System.NullReferenceException: Object reference not set to an instance of an object.

   at Avalonia.Controls.Primitives.PopupPositioning.PopupPositionerExtensions.ConfigurePosition(PopupPositionerParameters& positionerParameters, TopLevel topLevel, IVisual target, PlacementMode placement, Point offset, PopupAnchor anchor, PopupGravity gravity, PopupPositionerConstraintAdjustment constraintAdjustment, Nullable`1 rect) in /_/src/Avalonia.Controls/Primitives/PopupPositioning/IPopupPositioner.cs:line 502
   at Avalonia.Controls.Primitives.PopupRoot.ConfigurePosition(IVisual target, PlacementMode placement, Point offset, PopupAnchor anchor, PopupGravity gravity, PopupPositionerConstraintAdjustment constraintAdjustment, Nullable`1 rect) in /_/src/Avalonia.Controls/Primitives/PopupRoot.cs:line 95
   at Avalonia.Controls.ToolTip.Open(Control control) in /_/src/Avalonia.Controls/ToolTip.cs:line 283
   at Avalonia.Controls.ToolTip.IsOpenChanged(AvaloniaPropertyChangedEventArgs e) in /_/src/Avalonia.Controls/ToolTip.cs:line 241
   at System.Reactive.Subjects.Subject`1.OnNext(T value) in /_/Rx.NET/Source/src/System.Reactive/Subjects/Subject.cs:line 145
   at Avalonia.AvaloniaObject.RaisePropertyChanged[T](AvaloniaPropertyChangedEventArgs`1 change) in /_/src/Avalonia.Base/AvaloniaObject.cs:line 756
   at Avalonia.AvaloniaObject.Avalonia.PropertyStore.IValueSink.ValueChanged[T](AvaloniaPropertyChangedEventArgs`1 change) in /_/src/Avalonia.Base/AvaloniaObject.cs:line 544
   at Avalonia.ValueStore.NotifyValueChanged[T](AvaloniaProperty`1 property, Optional`1 oldValue, BindingValue`1 newValue, BindingPriority priority) in /_/src/Avalonia.Base/ValueStore.cs:line 366
   at Avalonia.ValueStore.SetExisting[T](Object slot, StyledPropertyBase`1 property, T value, BindingPriority priority) in /_/src/Avalonia.Base/ValueStore.cs:line 266
   at Avalonia.ValueStore.SetValue[T](StyledPropertyBase`1 property, T value, BindingPriority priority) in /_/src/Avalonia.Base/ValueStore.cs:line 100
   at Avalonia.Controls.ToolTip.SetIsOpen(Control element, Boolean value) in /_/src/Avalonia.Controls/ToolTip.cs:line 122
   at Avalonia.Interactivity.EventRoute.RaiseEventImpl(RoutedEventArgs e) in /_/src/Avalonia.Interactivity/EventRoute.cs:line 152
   at Avalonia.Interactivity.EventRoute.RaiseEvent(IInteractive source, RoutedEventArgs e) in /_/src/Avalonia.Interactivity/EventRoute.cs:line 79
   at Avalonia.Interactivity.Interactive.RaiseEvent(RoutedEventArgs e) in /_/src/Avalonia.Interactivity/Interactive.cs:line 123
   at Avalonia.Input.MouseDevice.SetPointerOver(IPointerDevice device, UInt64 timestamp, IInputRoot root, IInputElement element, PointerPointProperties properties, KeyModifiers inputModifiers) in /_/src/Avalonia.Input/MouseDevice.cs:line 488
   at Avalonia.Input.MouseDevice.SetPointerOver(IPointerDevice device, UInt64 timestamp, IInputRoot root, Point p, PointerPointProperties properties, KeyModifiers inputModifiers) in /_/src/Avalonia.Input/MouseDevice.cs:line 424
   at Avalonia.Input.MouseDevice.MouseMove(IMouseDevice device, UInt64 timestamp, IInputRoot root, Point p, PointerPointProperties properties, KeyModifiers inputModifiers, Lazy`1 intermediatePoints) in /_/src/Avalonia.Input/MouseDevice.cs:line 273
   at Avalonia.Input.MouseDevice.ProcessRawEvent(RawPointerEventArgs e) in /_/src/Avalonia.Input/MouseDevice.cs:line 144
   at Avalonia.Input.InputManager.ProcessInput(RawInputEventArgs e) in /_/src/Avalonia.Input/InputManager.cs:line 37
   at Avalonia.Native.WindowBaseImpl.RawMouseEvent(AvnRawMouseEventType type, UInt32 timeStamp, AvnInputModifiers modifiers, AvnPoint point, AvnVector delta) in /_/src/Avalonia.Native/WindowImplBase.cs:line 317
   at Avalonia.Native.Interop.Impl.__MicroComIAvnWindowBaseEventsVTable.RawMouseEvent(Void* this, AvnRawMouseEventType type, UInt32 timeStamp, AvnInputModifiers modifiers, AvnPoint point, AvnVector delta) in /_/src/Avalonia.Native/Interop.Generated.cs:line 3785
--- End of stack trace from previous location ---
   at Avalonia.Native.PlatformThreadingInterface.RunLoop(CancellationToken cancellationToken) in /_/src/Avalonia.Native/PlatformThreadingInterface.cs:line 90
   at Avalonia.Threading.Dispatcher.MainLoop(CancellationToken cancellationToken) in /_/src/Avalonia.Base/Threading/Dispatcher.cs:line 65
   at Avalonia.Controls.ApplicationLifetimes.ClassicDesktopStyleApplicationLifetime.Start(String[] args) in /_/src/Avalonia.Controls/ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime.cs:line 120
   at Avalonia.ClassicDesktopStyleApplicationLifetimeExtensions.StartWithClassicDesktopLifetime[T](T builder, String[] args, ShutdownMode shutdownMode) in /_/src/Avalonia.Controls/ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime.cs:line 209
   at Bulk_Edit_Calendar_Events_WPF.Program.Main(String[] args) in /Users/derekantrican/Public/Drop Box/Bulk Edit Calendar Events/Code/Bulk Edit Calendar Events_WPF/Program.cs:line 40
```

**Desktop (please complete the following information):**

- OS: OSX (Mac) - not sure if it's been on seen on Windows, seems to be only on Mac currently
- Version: 0.10.15.0

**Comment by maxkatz6** [comment]
I tried to find out line where this exception occurs, but it seem to not match 0.10.15 source code what we have: https://github.com/AvaloniaUI/Avalonia/blob/release/0.10.15/src/Avalonia.Controls/Primitives/PopupPositioning/IPopupPositioner.cs#L502

I guess the biggest chance to get NRE there is from first line:
```
var pointer = topLevel.PointToClient(topLevel.PlatformImpl.MouseDevice.Position);
```
where topLevel.PlatformImpl can be null, when avalonia window is already disposed, but event was raised after.

**Comment by maxkatz6** [comment]
In which case it should be fixed in 11.0 code. But can't confirm.
0.10.15/0.10.18: https://github.com/AvaloniaUI/Avalonia/blob/6239b3acf1e50c046cd2a6b137be1a30b5c56213/src/Avalonia.Controls/TopLevel.cs#L378
11.0 https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/TopLevel.cs#L394

11.0 https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/Primitives/PopupPositioning/IPopupPositioner.cs#L461
